src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Nls.cs

```diff
@@ -114,6 +117,13 @@ public partial class JapaneseCalendar : Calendar
             // Return our ranges
             return registryEraRanges;
         }
+#else
+        // no-op, in Unix we never call this function.
+        // the reason to have it is to simplify the build
+        // this way we avoid having to include RegistryKey
+        // and all it's windows PInvokes.
+        private static EraInfo[]? NlsGetJapaneseEras() => null;
```

Comment:
So then should the implementation instead be as below?

```cs
private static EraInfo[]? NlsGetJapaneseEras()
{
    Debug.Fail("Should never be called on non-Windows platforms.");
    throw new PlatformNotSupportedException();
}
```

Reply:
Yeah, good call.